src/app/public/modules/dynamic-component/dynamic-component.service.ts

```diff
@@ -90,6 +90,9 @@ export class SkyDynamicComponentService {
 
     const el = this.getRootNode(cmpRef);
 
+    // Call the destroy method to ensure that any ngOnDestroy methods are called
+    cmpRef.destroy();
```

Comment:
We should probably run the `detachView` method, too?
```
this.appRef.detachView(componentRef.hostView);
componentRef.destroy();
this.renderer.removeChild(bodyEl, el);
```

Reply:
Good call. Done